File: problem-solving/chunkarray/chunk-array.ts
const len = <T>(list: Array<T>) => list.length

const chunkArray = (list: number[], size: number) => {
  const result = []
  let n = 0
  while (n < len(list)) {
    result.push(list.slice(n, size + n))
    n += size
  }
  return result
}

const chunkArray2 = (list: number[], size: number) => {
  const fn = (list: number[], size: number) => {
    const result: number[][] = []

    for (let i = 0; i < list.length; i += size) {
      result.push(list.slice(i, i + size))
    }

    return result
  }
}

File: problem-solving/chunkarray/chunk-array.go
package chunkarray

// ChunkList func
func ChunkList(list []int, level int) [][]int {
	var result [][]int
	n := 0
	for n < len(list) {
		result = append(result, list[n:n+level])
		n += level
	}

	return result

}

// ChunkList2 func
func ChunkList2(list []int, level int) [][]int {
	var result [][]int
	for i := 0; i < len(list); i += level {
		result = append(result, list[i:level+i])
	}
	return result

}
